src/components/ShareSearch.js

```diff
@@ -0,0 +1,83 @@
+// @flow
+import React from 'react';
+import PropTypes from 'prop-types';
+import { withRouter } from 'react-router-dom';
+
+import Glyphicon from 'react-bootstrap/lib/Glyphicon';
+
+import Configurable from './Configurable';
+import AuthUtils from '../util/AuthUtils';
+
+type ShareSearchProps = {
+  history: PropTypes.object.isRequired,
+  /**
+   * The message body for when a search is shared.
+   */
+  shareMessage?: string,
+  /**
+   * Subject of the email for when a search is shared.
+   */
+  subject?: string,
+  /**
+   * Email address the search will be shared with when using shareSearch.
+   */
+  email?: string,
+};
+
+type ShareSearchDefaultProps = {
+  shareMessage: string,
+  subject: string,
+  email: string,
+};
+
+/**
+ * Component to include in the Masthead for entering the query
+ * to use when searching. Must be inside a Searcher component.
+ */
+class ShareSearch extends React.Component<ShareSearchDefaultProps, ShareSearchProps, void> {
+  static defaultProps: ShareSearchDefaultProps = {
+    shareMessage: `Hey,
+    
+    I think you would be interested in these search results that I found using Attivio, a leader in cognitive search and knowledge discovery. Here is the link:`,
+    subject: 'Search results I found using Attivio!',
+    email: '',
+  };
+
+  static contextTypes = {
+    searcher: PropTypes.any,
+  };
+
+  static displayName = 'ShareSearch';
+
+  render() {
+    const username = AuthUtils.getUserName(AuthUtils.getSavedUser());
```

Comment:
Prefer [destructuring](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment):
>**Default values**
>A variable can be assigned a default, in the case that the value unpacked from the object is undefined.
>**Assigning to new variables names and providing default values**
>A property can be both 1) unpacked from an object and assigned to a variable with a different name and 2) assigned a default value in case the unpacked value is undefined.
```
const {
  shareMessage: message,
  history: { location: { pathname: searchlink } }, // rename here if you like
  subject = '', // set a default value
  email: emailAddress = '', // rename and set a default value
} = this.props;
const username = AuthUtils.getUserName(AuthUtils.getSavedUser());
const signature = `\nfrom \n${username}`;
const emailBody = message + searchLink + signature;
```